### PR TITLE
Merge ToC and sidebar

### PR DESCRIPTION
**Pull request by joelfranusic-okta**
This PR attempts to combine #416 and #432

Changes include:
- Discarding the "hamburger" from #416 in favor of keeping the previous dropdown menu
- Updates to headings in select API documents from #432 - to fix ToC rendering 
- Disable `scrollLock()` on small screens

A preview of these changes is available at: https://frozen-ridge-22271.herokuapp.com/

### CODE REVIEW

**Comment by joelfranusic-okta** [comment]
@johngronberg-okta @karlmcguinness-okta @mystiberry-okta for your review.

**Comment by mystiberry-okta** [comment]
👍

**Comment by johngronberg-okta** [comment]
@karlmcguinness-okta @rchild-okta LGTM, but can you confirm?

File: _source/_sass/okta/components/_Sidebar.scss [hunk 1]
```diff
@@ -81,14 +81,14 @@
...
-@include media("<=medium") {
+@include media('<doc-min-toc') {
 	.Sidebar {
 		@include padding(x-small small x-small 0);
 		background: get-color('white');
 		min-width: 100%;
 		position: relative;
-		clear:both;
+		clear: both;
+        order: -2;
```
Comment (rchild-okta) [inline _source/_sass/okta/components/_Sidebar.scss]: nit: spacing

File: _source/assets/css/okta.scss [hunk 2]
```diff
@@ -27,6 +27,7 @@
 	'okta/components/Hero.scss',
 	'okta/components/SideBySidePromo.scss',
 	'okta/components/IconTiles.scss',
+    'okta/components/TableOfContents',
```
Comment (rchild-okta) [inline _source/assets/css/okta.scss]: nit: spacing

In general, I'm wondering if they're using tabs & we're using spaces (at some point we should switch to spaces and not have to worry about these kinds of issues)

File: _source/assets/js/docs.js [hunk 3]
```diff
@@ -43,9 +43,257 @@ $(function() {
 	});
 
 	if ($(location.href.split("#")[1])) {
-		var target = $('#' + location.href.split("#")[1]);
-		if (target.length) {
-			$('html,body').scrollTop(target.offset().top - fixedNavHeight);
+		var anchor = location.href.split("#")[1];
+			if (anchor) {
```
Comment (rchild-okta) [inline _source/assets/js/docs.js]: spacing

File: _source/assets/js/docs.js [hunk 4]
```diff
@@ -43,9 +43,257 @@ $(function() {
 	});
 
 	if ($(location.href.split("#")[1])) {
-		var target = $('#' + location.href.split("#")[1]);
-		if (target.length) {
-			$('html,body').scrollTop(target.offset().top - fixedNavHeight);
+		var anchor = location.href.split("#")[1];
```
Comment (rchild-okta) [inline _source/assets/js/docs.js]: location.hash instead of href (does it for us).

File: _source/assets/js/docs.js [hunk 5]
```diff
@@ -43,9 +43,257 @@ $(function() {
...
+			var target = $('#' + anchor);
+			if (target.length) {
+				$('html,body').scrollTop(target.offset().top - fixedNavHeight);
+			}
 		}
 	};
+
+
+
+	/*!
```
Comment (rchild-okta) [inline _source/assets/js/docs.js]: Would be nicer to move all vendor files to a specific vendor folder, and then include them on the page vs. including them inline with our source code.

File: _source/assets/js/docs.js [hunk 6]
```diff
@@ -43,9 +43,257 @@ $(function() {
...
+
+			$toc.append('<div class="TableOfContents-footer"><a href="#">Top of Page</a></div>');
+
+		};
+
+		var init = function() {
+			$page.addClass('has-tableOfContents');
+            // Defined in _sass/okta/components/_Sidebar.scss
+            // See also: _source/_sass/okta/_/base/_variables.scss:66
+            if ($(window).width() > 1024) {
```
Comment (rchild-okta) [inline _source/assets/js/docs.js]: spacing

**Comment by joelfranusic-okta** [comment]
@rchild-okta I addressed your formatting items, but I'm leaving your feedback on `doc.js` as-is since both of those items seem like good things to address as part of a larger refactor effort?

**Comment by karlmcguinness-okta** [comment]
I get compile errors with `rm -r _site/*; jekyll serve --incremental --verbose`

```
  Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/okta.scss':
                    Invalid CSS after "...conTiles.scss',": expected file to import (string or url()), was "okta/components..." on line 28
jekyll 3.2.1 | Error:  Invalid CSS after "...conTiles.scss',": expected file to import (string or url()), was "okta/components..." on line 28
```

```
karl@guinness: ~/okta/okta.github.io on merge-toc-and-sidebar [!$]
$ git log
commit 4065a9081917754eae0684d00ecf83e34ff5e9c2
Author: Joel Franusic <joel.franusic@okta.com>
Date:   Tue Oct 25 17:18:01 2016 -0700

    Address rchild-okta review items

```

```
$ git status
On branch merge-toc-and-sidebar
Your branch is up-to-date with 'origin/merge-toc-and-sidebar'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Gemfile.lock

no changes added to commit (use "git add" and/or "git commit -a")
```

File: _source/assets/css/okta.scss [hunk 7]
```diff
@@ -27,10 +27,11 @@
 	'okta/components/Hero.scss',
 	'okta/components/SideBySidePromo.scss',
 	'okta/components/IconTiles.scss',
-	
+	okta/components/TableOfContents',
```
Comment (johngronberg-okta) [inline _source/assets/css/okta.scss]: looks like this is missing a ' 

should be 'okta/components/TableOfContents',

**Comment by johngronberg-okta** [comment]
ok- works for me after the latest commit. LGTM

**Comment by mystiberry-okta** [comment]
@joelfranusic-okta okay to merge and publish?

**Pull request merged**